src/Plugins/ComplexCore/test/ChangeAngleRepresentationTest.cpp

```diff
@@ -0,0 +1,126 @@
+#include "complex/UnitTest/UnitTestCommon.hpp"
+
+using namespace complex;
+
+TEST_CASE("OrientationAnalysis::ChangeAngleRepresentation: Instantiation and Parameter Check", "[OrientationAnalysis][ChangeAngleRepresentation]")
+{
+  // Instantiate the filter, a DataStructure object and an Arguments Object
+  ChangeAngleRepresentation filter;
+  DataStructure ds;
+  Arguments args;
+
+  // Create default Parameters for the filter.
+  // This should fail
+  args.insertOrAssign(ChangeAngleRepresentation::k_ConversionType_Key, std::make_any<ChoicesParameter::ValueType>(0));
+  args.insertOrAssign(ChangeAngleRepresentation::k_AnglesArrayPath_Key, std::make_any<DataPath>(DataPath{}));
+
+  // Preflight the filter and check result
+  auto preflightResult = filter.preflight(ds, args);
+  REQUIRE(preflightResult.outputActions.valid() == false);
+
+  // This should fail because parameter is out of range
+  args.insertOrAssign(ChangeAngleRepresentation::k_ConversionType_Key, std::make_any<ChoicesParameter::ValueType>(2));
+  preflightResult = filter.preflight(ds, args);
+  REQUIRE(preflightResult.outputActions.valid() == false);
+}
+
+TEST_CASE("OrientationAnalysis::ChangeAngleRepresentation: Degrees To Radians")
+{
+  // Instantiate the filter, a DataStructure object and an Arguments Object
+  ChangeAngleRepresentation filter;
+  DataStructure dataGraph;
+  Arguments args;
+
+  DataGroup* topLevelGroup = DataGroup::Create(dataGraph, Constants::k_SmallIN100);
+  DataGroup* scanData = DataGroup::Create(dataGraph, Constants::k_EbsdScanData, topLevelGroup->getId());
+
+  std::vector<size_t> tupleShape = {10};
+  std::vector<size_t> componentShape = {3};
+
+  Float32Array* angles = UnitTest::CreateTestDataArray<float>(dataGraph, Constants::k_EulerAngles, tupleShape, componentShape, scanData->getId());
+
+  for(size_t t = 0; t < tupleShape[0]; t++)
+  {
+    for(size_t c = 0; c < componentShape[0]; c++)
+    {
+      (*angles)[t * componentShape[0] + c] = static_cast<float>(t * c);
+    }
+  }
+
+  // Create default Parameters for the filter.
+  // This should fail
+  args.insertOrAssign(ChangeAngleRepresentation::k_ConversionType_Key, std::make_any<ChoicesParameter::ValueType>(0));
+  args.insertOrAssign(ChangeAngleRepresentation::k_AnglesArrayPath_Key, std::make_any<DataPath>(DataPath({Constants::k_SmallIN100, Constants::k_EbsdScanData, Constants::k_EulerAngles})));
+
+  // Preflight the filter and check result
+  auto preflightResult = filter.preflight(dataGraph, args);
+  REQUIRE(preflightResult.outputActions.valid());
+
+  // Execute the filter and check the result
+  auto executeResult = filter.execute(dataGraph, args);
+  REQUIRE(executeResult.result.valid());
+
+  // Check the results
+  float d2r = complex::numbers::pi / 180.0f;
+  for(size_t t = 0; t < tupleShape[0]; t++)
+  {
+    for(size_t c = 0; c < componentShape[0]; c++)
+    {
+      REQUIRE((*angles)[t * componentShape[0] + c] == static_cast<float>(t * c) * d2r);
```

Comment:
We should typically keep `REQUIRE` out of loops since it can be much slower. Though since there are only 30 elements in this case, it shouldn't have much of an impact.

Reply:
Exactly. I'm taking this case by case which is why I left it in the loop.